### DISCUSSION TITLE
Packing multiple packets to 1 write

### DISCUSSION THREAD

**Discussion by andreasdc** [netty/netty#11911] [Q&A]
Hey,
I was suggested by Norman Maurer to pack multiple packets into 1 write call. His original idea was:
Pack multiple messages into one "object" pass it to writeAndFlush(...) and have your own ChannelOutboundHandler that "decompose" this object into messages.
How to achieve that? Will client get every packet as 1 object or the whole package of packets? It needs to be every packet as a single object.

**Comment by chrisvest** [comment]
Since we're talking about packets I assume you're writing to a `DatagramChannel`. In that case, if you use the `epoll` transport, then multiple writes will be collected and sent with `sendmmsg` on channel `flush`. Additionally, any `SegmentedDatagramPacket` will also be sent with `sendmmsg`, but for these all the segments will have the same size and come from the same buffer.

**Reply by andreasdc** [reply]
> Since we're talking about packets I assume you're writing to a `DatagramChannel`. In that case, if you use the `epoll` transport, then multiple writes will be collected and sent with `sendmmsg` on channel `flush`. Additionally, any `SegmentedDatagramPacket` will also be sent with `sendmmsg`, but for these all the segments will have the same size and come from the same buffer.

The problem is that I want to get rid of waking up netty and improve performance of sending packets.

**Reply by chrisvest** [reply]
Make sure you do your writes from within the event loop thread.

**Reply by andreasdc** [reply]
> Make sure you do your writes from within the event loop thread.

I tried experimenting with outside writing, but I didn't measure it yet. There is no big difference for sure. 

I mean is it possible to write something like a list of packets, then split it to single packets? Client must receive every packet one by one.
